File: cpp/1903_largest_odd_number_in_string.cpp
/*
https://leetcode.com/problems/largest-odd-number-in-string/submissions/1114563312/
*/

class Solution {
public:
    string largestOddNumber(string num) {
        int end = -1;
        for (int i = num.length() - 1; i > -1; i--) {
            if ((num[i] - 0) % 2 != 0) {
                end = i;
                break;
            }
        }
        if (end == -1) {
            return "";
        }
        return num.substr(0, end + 1);
    }
};

File: python/1903_largest_odd_number_in_string.py
'''
https://leetcode.com/problems/largest-odd-number-in-string/submissions/1245981975/
'''


class Solution:
    def largestOddNumber(self, num: str) -> str:
        i = len(num) - 1
        while (i > -1):
            if (int(num[i]) % 2 == 1):
                break
            i -= 1
        return num[0:i+1]